src/http/server/handlers/static_file_handler/cache.cr

```diff
@@ -0,0 +1,41 @@
+module HTTP::StaticFileHandler::Cache
+  HTTP_IF_MODIFIED_SINCE = "If-Modified-Since"
+  HTTP_IF_NONE_MATCH     = "If-None-Match"
+  HTTP_ETAG              = "Etag"
+  HTTP_LAST_MODIFIED     = "Last-Modified"
+
+  def self.add_headers(context : HTTP::Server::Context, file_path : String) : Nil
+    context.response.headers[HTTP_ETAG] = etag(file_path)
+
+    last_modified = modification_time(file_path)
+    context.response.headers[HTTP_LAST_MODIFIED] = HTTP.rfc1123_date(last_modified)
+  end
+
+  def self.match?(context : HTTP::Server::Context, file_path : String) : Bool
+    if_none_match = context.request.headers[HTTP_IF_NONE_MATCH]?
+
+    # According to RFC 7232:
+    # A recipient must ignore If-Modified-Since if the request contains an If-None-Match header field
+    return if_none_match == context.response.headers[HTTP_ETAG] if if_none_match
+
+    if_modified_since = context.request.headers[HTTP_IF_MODIFIED_SINCE]?
+    return false unless if_modified_since
+
+    header_time = HTTP.parse_time(if_modified_since)
+
+    # File mtime probably has a higher resolution than the header value.
+    # An exact comparison might be slightly off, so we add 1s padding.
+    # Static files should generally not be modified in subsecond intervals, so this is perfectly safe.
+    # This might be replaced by a more sophisticated time comparison when it becomes available.
+    last_modified = modification_time(file_path)
+    !!(header_time && last_modified <= header_time + 1.second)
+  end
+
+  private def self.etag(file_path)
+    %{W/"#{File.info(file_path).modification_time.epoch.to_s}"}
```

Comment:
Don't need `.to_s` here. In fact, it results in a additional, useless string allocation.